runner/src/runner.rs

```diff
@@ -157,45 +340,48 @@ impl RunnerExt for Runner {
     }
 
     async fn try_get_release(&self, pending: bool) -> Result<Option<ClientRelease>, Error> {
-        try_get_release(self, pending).await
+        Runner::try_get_release(self, pending).await
     }
 
-    fn run_binary_with_retry(&mut self) -> Result<(), Error> {
-        let child = run_binary_with_retry(self, Stdio::inherit())?;
-        self.set_child_proc(child);
+    fn run_binary(&mut self) -> Result<(), Error> {
+        let child = Runner::run_binary(self, Stdio::inherit())?;
+        self.child_proc = Some(child);
         Ok(())
     }
 
     async fn download_binary(&mut self, release: ClientRelease) -> Result<(), Error> {
-        let downloaded_release = download_binary(&*self, release).await?;
-        self.set_downloaded_release(downloaded_release);
+        let _downloaded_release = Runner::download_binary(self, release).await?;
         Ok(())
     }
 
     fn uri_to_bin_path(&self, uri: &str) -> Result<(String, PathBuf), Error> {
-        uri_to_bin_path(self, uri)
+        Runner::uri_to_bin_path(self, uri)
     }
 
     fn delete_downloaded_release(&mut self) -> Result<(), Error> {
-        delete_downloaded_release(self)?;
-        self.downloaded_release = None;
+        Runner::delete_downloaded_release(self)?;
         Ok(())
     }
 
     fn terminate_proc_and_wait(&mut self) -> Result<(), Error> {
-        terminate_proc_and_wait(self)?;
-        self.child_proc = None;
+        Runner::terminate_proc_and_wait(self)?;
         Ok(())
     }
 
     // Declaring as a static method would highly complicate mocking
     async fn get_request_bytes(&self, url: String) -> Result<Bytes, Error> {
-        let response = reqwest::get(url.clone()).await?;
-        Ok(response.bytes().await?)
+        log::info!("Fetching executable from {}", url);
+        Ok(retry_with_log_async(
```

Comment:
Trying to think how best to approach this API, as a consumer given the method name "get_request_bytes" I would not expect it to retry internally - rather the consumer of the call should do that. However, I'm just being nitpicky here as this only affects the internal implementation anyway so just food for thought...

Reply:
Good point, I moved the retry to the calling function